### PR TITLE
Fix Gentoo hostname bug

### PR DESCRIPTION
**Pull request by manuelisimo**
Gentoo's hostname file format instead of being just the host name by itself is `hostname="thename"`. The existing code works fine when the file has no comments but if there is a comment, then the line:

```
gentoo_hostname_config = 'hostname="%s"' % conf
```

can render an invalid hostname file that looks similar to

```
hostname="#This is the host namehello"
```
The fix calls `set_hostname` with the parameter in a gentoo friendly way so that it gets handled by HostnameConf as a whole, and comments are handled and preserved

### CODE REVIEW

**Review by otubo — APPROVED** [review]
It does look good to me.

**Comment by manuelisimo** [comment]
Re-based branch and fixed flake8 issue

**Review by TheRealFalcon — APPROVED** [review]

File: tools/.github-cla-signers [hunk 1]
```diff
@@ -13,6 +13,7 @@ johnsonshi
 jqueuniet
 landon912
 lucasmoura
+manuelisimo
```
Comment (OddBloke) [inline tools/.github-cla-signers:16]: I've confirmed this signature internally, thanks!

File: cloudinit/distros/gentoo.py [hunk 2]
```diff
@@ -160,10 +160,12 @@ def _write_hostname(self, your_hostname, out_fn):
...
         if not conf:
             conf = HostnameConf('')
-        conf.set_hostname(your_hostname)
-        gentoo_hostname_config = 'hostname="%s"' % conf
-        gentoo_hostname_config = gentoo_hostname_config.replace('\n', '')
-        util.write_file(out_fn, gentoo_hostname_config, 0o644)
+
+        # Many distro's format is the hostname by itself, and that is the
+        # way HostnameConf works but gentoo expects it to be in
+        #     hostname="the-actual-hostname"
```
Comment (OddBloke) [inline cloudinit/distros/gentoo.py:166]: Confirmed at https://wiki.gentoo.org/wiki/Handbook:AMD64/Installation/System

**Review by OddBloke — APPROVED** [review]
LGTM, thanks!

**Pull request merged**